src/zkscript/merkle_trees/merkle_tree.py

```diff
@@ -0,0 +1,205 @@
+import string
+
+from tx_engine import Script
+
+
+class MerkleTree:
+    """Class implementing methods to generate locking/unlocking scripts verifying Merkle paths."""
+
+    def __init__(self, root: str, hash_function: str, depth: int):
+        """Initialize a MerkleTree instance.
+
+        Args:
+            root (str): The root hash of the Merkle Tree, provided as a hexadecimal string.
+            hash_function (str): Hash function used in the Merkle Tree, as a valid opcode or combination of opcodes.
+            depth (int): Number of levels in the Merkle tree.
+
+        Raises:
+            AssertionError: If `root` is not a hexadecimal or `hash_function` contains invalid opcodes.
+
+        """
+        assert all(c in string.hexdigits for c in root), f"{root} is not a valid hexadecimal string."
+
+        assert set(hash_function.split(" ")).issubset(
+            {"OP_RIPEMD160", "OP_SHA1", "OP_SHA256", "OP_HASH160", "OP_HASH256"}
+        ), f"{hash_function} is not a valid hash function."
+
+        assert depth > 0
+
+        self.root = root
+        self.hash_function = hash_function
+        self.depth = depth
+
+    def unlocking_merkle_proof_with_bit_flags(
+        self,
+        d: str,
+        aux: list[str],
+        bit: list[bool],
+    ) -> Script:
+        """Generate unlocking scripts for Merkle path verification using a bit flag to identify the right and left path.
+
+        The unlocking script loads the data on the stack, sorthing them as following:
+        [... aux_i bit_i ... aux_1 bit_1 d]
+
+        Args:
+            d (`str`): Data for which the Merkle root is being proved, as a hexadecimal string.
```

Comment:
Remove the ` in the types of arguments throughout

Reply:
done